src/utils/rpg-system/rpgSystem.ts

```diff
@@ -2,6 +2,8 @@ import { DEFAULT_UNDEFINED } from "../constants";
 import type { GenericCreature, DifficultyLevel, DifficultyThreshold } from "./index";
 
 export abstract class RpgSystem {
+  public abstract SystemDifficulties: string[];
```

Comment:
Can you define with a default value, so that it doesn't throw an error if there's an issue with the config or the rpgSystem can't be found?

Also:
- if this is used for the meter, I think we want to enforce via typing that there are at least two values.
- `public` isn't necessary; it's the default visibility. better to leave it off for consistency
- should be named with a lowercase first letter

e.g.
```
systemDifficulties: [string, string, ...string[]] = [DEFAULT_UNDEFINED, DEFAULT_UNDEFINED];
```

Reply:
Made the changes, but left the parent property abstract and added the default values to `UndefinedRpgSystem`